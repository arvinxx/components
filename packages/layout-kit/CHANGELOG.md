# Changelog

## @arvinxu/layout-kit [1.2.1](https://github.com/arvinxx/components/compare/@arvinxu/layout-kit@1.2.0...@arvinxu/layout-kit@1.2.1) (2022-02-03)

### 🎫 杂项

- 从 yarn 迁移到 pnpm ([38bb09b](https://github.com/arvinxx/components/commit/38bb09b))

### 👷 构建系统

- 发布新版 ([40b74d4](https://github.com/arvinxx/components/commit/40b74d4))

# @arvinxu/layout-kit [1.2.0](https://github.com/arvinxx/components/compare/@arvinxu/layout-kit@1.1.0...@arvinxu/layout-kit@1.2.0) (2022-02-03)

### ✨ 新特性

- 针对横向的 flexbox，默认将宽度撑满 ([e41c6a2](https://github.com/arvinxx/components/commit/e41c6a2))

# @arvinxu/layout-kit [1.1.0](https://github.com/arvinxx/components/compare/@arvinxu/layout-kit@1.0.0...@arvinxu/layout-kit@1.1.0) (2022-02-02)

### ✅ 测试

- 修正测试 ([74e0652](https://github.com/arvinxx/components/commit/74e0652))

### ✨ 新特性

- 新增 horizontal props ([1f38438](https://github.com/arvinxx/components/commit/1f38438))

### 🎫 杂项

- **自动发布**: @arvinxu/layout-kit@1.1.0-beta.1 [skip ci] ([8c86eb5](https://github.com/arvinxx/components/commit/8c86eb5))
- 锁定 react-router 版本 ([e913677](https://github.com/arvinxx/components/commit/e913677))

### 📝 文档

- 更新文档 ([cca4137](https://github.com/arvinxx/components/commit/cca4137))

# @arvinxu/layout-kit [1.1.0-beta.1](https://github.com/arvinxx/components/compare/@arvinxu/layout-kit@1.0.0...@arvinxu/layout-kit@1.1.0-beta.1) (2022-02-02)

### ✅ 测试

- 修正测试 ([74e0652](https://github.com/arvinxx/components/commit/74e0652))

### ✨ 新特性

- 新增 horizontal props ([1f38438](https://github.com/arvinxx/components/commit/1f38438))

### 🎫 杂项

- 锁定 react-router 版本 ([e913677](https://github.com/arvinxx/components/commit/e913677))

### 📝 文档

- 更新文档 ([cca4137](https://github.com/arvinxx/components/commit/cca4137))

# @arvinxu/layout-kit 1.0.0 (2021-11-28)

### ✅ 测试

- fix test ([4101a41](https://github.com/arvinxx/components/commit/4101a41))

### ✨ 新特性

- 初步完成 Flexbox 组件 ([bfd8393](https://github.com/arvinxx/components/commit/bfd8393))

### 🎫 杂项

- rename package ([fa558eb](https://github.com/arvinxx/components/commit/fa558eb))
