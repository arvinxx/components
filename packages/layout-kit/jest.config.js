const base = require('../../jest.config.base');

const packageName = '@arvinxu/layout-kit';

const root = '<rootDir>/packages/layout-kit';

module.exports = {
  ...base,
  rootDir: '../..',
  roots: [root],
  name: packageName,
  displayName: packageName,
  collectCoverageFrom: [`${root}/src/**/*.tsx`, `${root}/src/**/*.ts`],
};
