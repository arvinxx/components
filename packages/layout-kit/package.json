{
  "name": "@arvinxu/layout-kit",
  "version": "1.1.0",
  "files": [
    "lib",
    "es"
  ],
  "main": "lib/index.js",
  "module": "es/index.js",
  "homepage": "https://github.com/arvinxx/components/tree/master/packages/layout-kit#readme",
  "repository": "git+https://github.com/arvinxx/components.git",
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "dependencies": {
    "styled-components": "^5.3.3"
  },
  "devDependencies": {
    "styled-components": "^5.3.3",
    "@types/styled-components": "^5.1.15"
  },
  "scripts": {
    "build": "father-build && yarn webpack",
    "webpack": "webpack",
    "test": "jest",
    "test:update": "jest -u",
    "prepublishOnly": "pnpm run build",
    "cov": "jest --coverage",
    "clean": "rm -rf es lib dist build coverage .umi"
  }
}
