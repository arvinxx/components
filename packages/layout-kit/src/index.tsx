import type { DetailedHTMLProps, FC, HTMLAttributes } from 'react';
import styled from 'styled-components';

import { calcValue, getFlexDirection } from './utils';
import type {
  ContentDistribution,
  ContentPosition,
  FlexDirection,
} from './type';

type CommonSpaceNumber = 2 | 4 | 8 | 12 | 16 | 24;

interface IFlexbox {
  direction?: FlexDirection;
  distribution?: ContentDistribution;
  align?: ContentPosition;
  gap?: CommonSpaceNumber | number;
  width?: number | string;
  height?: number | string;
  padding?: string | number | CommonSpaceNumber;
}
export type FlexboxProps = IFlexbox &
  DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>;

export const Flexbox: FC<FlexboxProps> = styled.div.attrs(() => ({
  className: 'layoutkit-flexbox',
}))<IFlexbox>`
  display: flex;
  flex-direction: ${(props) => {
    return getFlexDirection(props.direction);
  }};
  justify-content: ${(props) => props.distribution};
  align-items: ${(props) => props.align};

  width: ${(props) => calcValue(props.width)};
  height: ${(props) => calcValue(props.height)};

  padding: ${(props) => calcValue(props.padding)};

  > *:not(:last-child) {
    margin-right: ${(props) =>
      getFlexDirection(props.direction) === 'row' && calcValue(props.gap)};
    margin-bottom: ${(props) =>
      getFlexDirection(props.direction) === 'column' && calcValue(props.gap)};
  }
`;
