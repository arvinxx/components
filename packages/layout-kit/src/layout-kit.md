---
title: LayoutKit 布局工具库
group:
  path: /
nav:
  path: /components
---

# LayoutKit 布局工具库

[![NPM version][version-image]][version-url] [![NPM downloads][download-image]][download-url]

<!-- npm url -->

[version-image]: http://img.shields.io/npm/v/@arvinxu/layout-kit.svg?color=deepgreen&label=latest
[version-url]: http://npmjs.org/package/@arvinxu/layout-kit
[download-image]: https://img.shields.io/npm/dm/@arvinxu/layout-kit.svg
[download-url]: https://npmjs.org/package/@arvinxu/layout-kit

<API src='./index.tsx'></API>
