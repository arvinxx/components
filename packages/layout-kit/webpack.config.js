const path = require('path');
const config = require('../../webpack.config');

module.exports = {
  ...config,
  output: {
    ...config.output,
    library: 'LayoutKit',
    path: path.resolve(__dirname, 'dist'),
  },
};
