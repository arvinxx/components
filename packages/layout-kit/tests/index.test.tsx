import React from 'react';
import { render } from '@testing-library/react';

import { Flexbox } from '@arvinxu/layout-kit';

describe('Flexbox', () => {
  it('gap', () => {
    const { container } = render(
      <Flexbox gap={2}>
        <div>1</div>
        <div>2</div>
      </Flexbox>,
    );
    expect(container).toMatchSnapshot();
  });
});
