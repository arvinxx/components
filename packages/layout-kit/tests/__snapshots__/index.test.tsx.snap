// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexbox gap 1`] = `
<div>
  <div
    class="sc-bdvvtL nqSUR layoutkit-flexbox"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;
